classes/CartRule.php

```diff
@@ -979,8 +982,18 @@ public function getContextualValue($use_tax, Context $context = null, $filter =
 
         $all_cart_rules_ids = $context->cart->getOrderedCartRulesIds();
 
-        $cart_amount_ti = $context->cart->getOrderTotal(true, Cart::ONLY_PRODUCTS);
-        $cart_amount_te = $context->cart->getOrderTotal(false, Cart::ONLY_PRODUCTS);
+        if (!array_key_exists($context->cart->id, $cartAmountCache)) {
+            if (Tax::excludeTaxeOption()) {
+                $cartAmountCache[$context->cart->id]['te'] = $context->cart->getOrderTotal(false, Cart::ONLY_PRODUCTS);
+                $cartAmountCache[$context->cart->id]['ti'] = $cartAmountCache[$context->cart->id]['te'];
```

Comment:
Hmm not sure about this line. This is tempting to make the shortcut if the tax is excluded, but did you make sure the results are the same? It may have an impact on the shipping price.

Reply:
Yes I've checked inside the getOrderTotal(). If ONLY_PRODUCTS is used, the $with_taxes is changed to false if Tax::excludeTaxeOption is set, and we don't use it before.

Reply:
It only has an impact in ONLY_DISCOUNTS, BOTH and ONLY_SHIPPING mode

Reply:
We thought we did because the variable can be found a few lines before.
![capture du 2017-10-11 11-21-56](https://user-images.githubusercontent.com/6768917/31435148-96404332-ae76-11e7-8b62-aec9f22b6ac6.png)

But if we have a look at the tests, it seems good.